File: Ivan_Jimenez_Mastermind/core/src/main/java/com/example/ivanmastermind/MastermindGame.java
package com.example.ivanmastermind;

import com.badlogic.gdx.Game;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.glutils.ShapeRenderer;
import com.badlogic.gdx.assets.AssetManager;

public class MastermindGame extends Game {
    public static final int VIRTUAL_WIDTH = 800;
    public static final int VIRTUAL_HEIGHT = 600;

    private SpriteBatch batch;
    private ShapeRenderer shapeRenderer;
    private AssetManager assetManager;

    @Override
    public void create() {
        batch = new SpriteBatch();
        shapeRenderer = new ShapeRenderer();
        assetManager = new AssetManager();

        // Setup viewport and camera if needed

        // Load assets here if needed
        // assetManager.load("textures/peg.png", Texture.class);
        // assetManager.finishLoading();

        setScreen(new GameScreen(this));
    }

    public SpriteBatch getBatch() { return batch; }
    public ShapeRenderer getShapeRenderer() { return shapeRenderer; }
    public AssetManager getAssetManager() { return assetManager; }

    @Override
    public void dispose() {
        batch.dispose();
        shapeRenderer.dispose();
        assetManager.dispose();
        getScreen().dispose();
    }
}

File: mistermind/core/src/main/java/com/badlogic/drop/MastermindGame.kt
package com.badlogic.drop

import com.badlogic.gdx.Game
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.graphics.g2d.BitmapFont

class MastermindGame : Game() {
    lateinit var batch: SpriteBatch
    lateinit var font: BitmapFont

    override fun create() {
        batch = SpriteBatch()
        font = BitmapFont().apply {
            data.setScale(2.5f)  // Hacer el texto un poco más grande
        }
        setScreen(GameScreen(this))
    }

    override fun render() {
        super.render()
    }

    override fun dispose() {
        batch.dispose()
        font.dispose()
        screen?.dispose()
    }
}
